src/core/lib/support/string.c

```diff
@@ -279,7 +279,8 @@ int gpr_stricmp(const char *a, const char *b) {
 void *gpr_memrchr(const void *s, int c, size_t n) {
   if (s == NULL) return NULL;
   char *b = (char *)s;
-  for (size_t i = 0; i < n; i++) {
+  size_t i;
```

Comment:
I think we use this pattern in many places. Is this a problem? The source is not c89 :)